src/main/java/carpettisaddition/mixins/carpet/tweaks/command/playerActionEnhanced/EntityPlayerActionPackActionMixin.java

```diff
@@ -90,15 +92,24 @@ private void perTickMultiplier(EntityPlayerActionPack actionPack, EntityPlayerAc
 	{
 		if (this.perTick != null)
 		{
+			ServerPlayerEntity player = ((EntityPlayerActionPackAccessor)actionPack).getPlayer();
+			IServerPlayerEntity iplayer = (IServerPlayerEntity)(Object)player;
+			EntityPlayerActionPackActionTypeAccessor typeAccessor = ((EntityPlayerActionPackActionTypeAccessor)(Object)type);
+			EntityPlayerActionPack.Action self = (EntityPlayerActionPack.Action)(Object)this;
+			IEntityPlayerActionPackActionTypeUse actionTypeUSE = (IEntityPlayerActionPackActionTypeUse)(Object)EntityPlayerActionPack.ActionType.USE;
+
+			iplayer.swapOldPosRot(true);
+
 			for (int i = 0; i < this.perTick - 1; i++)
 			{
-				EntityPlayerActionPackActionTypeAccessor typeAccessor = ((EntityPlayerActionPackActionTypeAccessor)(Object)type);
-				ServerPlayerEntity player = ((EntityPlayerActionPackAccessor)actionPack).getPlayer();
-				EntityPlayerActionPack.Action self = (EntityPlayerActionPack.Action)(Object)this;
-
+				actionTypeUSE.setTickPart((i + 1) / (float)perTick);
 				typeAccessor.invokeExecute(player, self);
 				typeAccessor.invokeInactiveTick(player, self);
 			}
+
+			actionTypeUSE.setTickPart(1.0f);
+			iplayer.swapOldPosRot(false);
+			iplayer.pushOldPosRot();
```

Comment:
So here's where you record the rotation and position of the player:

- when the action starts
- when the action completes an execution

You're assuming that the player moves smoothly between 2 positions, but this assution might be wrong. The player might be teleported, and teleporation do not have interpolation

Reply:
This is true. I had largely discounted it as an issue originally because player reach meant the use attempts would do nothing anyway, but it will potentially load chunks which does suck.
Will have a think about it, seems like it will be a few more injects at least though, I'm not convinced it is worth handling this explicitly.

Reply:
imo, it's import to ensure interpolation only applies in the correct scenarios. As you mentioned, interpolating in unexpected scenarios can cause problems, especially since carpet doesn't provide fine-granular permission system for these commands

and that's why the interpolation feature is hard to implement